### ISSUE TITLE
! LaTeX Error: Environment CSLReferences undefined.

### ISSUE THREAD

**Issue by mkapur** [nwfsc-assess/sa4ss#16]
When running the example, I get the following:
```
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS _main.utf8.md --to latex --from markdown+autolink_bare_uris+tex_math_single_backslash --output _main.tex --lua-filter "C:\Users\mkapur\Documents\R\win-library\4.0\bookdown\rmarkdown\lua\custom-environment.lua" --lua-filter "C:\Users\mkapur\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "C:\Users\mkapur\Documents\R\win-library\4.0\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --table-of-contents --toc-depth 2 --template "C:\Users\mkapur\Documents\R\win-library\4.0\sa4ss\rmarkdown\templates\sa\resources\sadraft.tex" --number-sections --highlight-style tango --pdf-engine lualatex --top-level-division=section --wrap=none --default-image-extension=png --lua-filter=tagged-filter.lua --variable tables=yes --standalone -Mhas-frontmatter=false --citeproc 
! LaTeX Error: Environment CSLReferences undefined.
```
I see that on line ~176 of `_main.tex` there is a reference to [this rmakrdown issue](https://github.com/rstudio/rmarkdown/issues/1649) though I am not clear whether the entirety of the template fix is actually here. In any case, it apepars that the `.tex` file gets overwritten (from the cloud?) when I call `bookdown::render_book` so I wasn't able to test for myself whether putting in the following (which is just a few extra lines) might help things:
```
$if(csl-refs)$
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {$if(csl-hanging-indent)$\setlength{\parindent}{0pt}%
  \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces$endif$}%
  {\par}
$endif$
```

**Comment by mkapur** [comment]
Ah, that might be it. Based on the pandoc errors I'd received (and mentioned [here](#15 )) I installed the latest version (2.1.2). Do you know which version number is compatible?

**Comment by mkapur** [comment]
Update: after some trial & error, I was ble to get the PDF to generate (though it added this warning:
```
Warning messages:
1: In has_crop_tools() : 
Tool(s) not installed or not in PATH: ghostscr
```

To do so, I downloaded `[pandoc 2.8.1](https://github.com/jgm/pandoc/releases/tag/2.8.1)`. (The version right before this, `2.8.0.1` had bibliography issues). In the process I realized that RStudio does not always update the version of pandoc used based on the most recent install, so I had to manually copy the downloaded .exe from AppData\Local\Pandoc to RStudio\bin\pandoc.

Can close unless there's anything else you want to add - thanks!
